app/Main.hs

```diff
@@ -130,17 +130,29 @@ main = do
   newEnvOrErrors <- Async.mapConcurrently (requestSecret opts) secrets
 
   let
+    checkNoDuplicates e =
+      let keys = map fst (e ++ env)
+      in case dups keys of
+           Left x -> errorWithoutStackTrace $ "Found duplicate variable \"" ++ x ++ "\""
```

Comment:
Justa driveby comment. @duijf perhaps we cant to switchj away from `errorWithoutStackTrace` and move to `ExceptT/Either` ?  Partial functions are never fun.

Reply:
We can look into that! I do prefer merging this PR first though

Reply:
Right now, we do use `Either` and only run `errorWithoutStackTrace` in `main`. That won't change if we put this logic in `Either`/`ExceptT`, right? (Can you open an issue where you make the case for this if you think this is important?)

Reply:
The case is that partial functions are always evil :tm: but i'll make a separate issue for it :)